ipa/scripts/cli_utils.py

```diff
@@ -20,13 +20,52 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import os
 import shlex
+import shutil
 import sys
 
 import click
 
 from ipa.ipa_controller import collect_results
-from ipa.ipa_utils import parse_test_name
+from ipa.ipa_utils import parse_test_name, update_history_log
+
+
+def archive_history_item(item, destination):
+    """
+    Archive the log and results file for the given history item.
+
+    Copy the files and update the results file in destination directory.
+    """
+    log_src, description = split_history_item(item.strip())
+
+    # Get relative path for log:
+    # {provider}/{image}/{instance}/{timestamp}.log
+    log_dest = os.path.sep.join(log_src.rsplit(os.path.sep, 4)[1:])
+
+    # Get results src and destination based on log paths.
+    results_src = log_src.rsplit('.', 1)[0] + '.results'
+    results_dest = log_dest.rsplit('.', 1)[0] + '.results'
+
+    try:
+        destination_path = os.path.join(destination, log_dest)
+
+        log_dir = os.path.dirname(destination_path)
+        if not os.path.isdir(log_dir):
+            os.makedirs(log_dir)
+
+        # Copy results and log files to archive directory.
+        shutil.copyfile(log_src, destination_path)
+        shutil.copyfile(results_src, os.path.join(destination, results_dest))
+    except Exception:
+        pass  # File does not exist, nothing we can do now.
```

Comment:
Should log or print an info message that tells the user that the selected item was not found and based on the "else" clause we should abandon the archive operation at this point all together, more below.